option.go

```diff
@@ -0,0 +1,123 @@
+package bmclib
+
+import (
+	"context"
+	"crypto/x509"
+	"net/http"
+	"time"
+
+	"github.com/bmc-toolbox/bmclib/v2/internal/httpclient"
+	"github.com/go-logr/logr"
+	"github.com/jacobweinstock/registrar"
+)
+
+// Option for setting optional Client values
+type Option func(*Client)
+
+// WithLogger sets the logger
+func WithLogger(logger logr.Logger) Option {
+	return func(args *Client) { args.Logger = logger }
+}
+
+// WithRegistry sets the Registry
+func WithRegistry(registry *registrar.Registry) Option {
+	return func(args *Client) { args.Registry = registry }
+}
+
+// WithSecureTLS enforces trusted TLS connections, with an optional CA certificate pool.
+// Using this option with an nil pool uses the system CAs.
+func WithSecureTLS(rootCAs *x509.CertPool) Option {
+	return func(args *Client) {
+		args.httpClientSetupFuncs = append(args.httpClientSetupFuncs, httpclient.SecureTLSOption(rootCAs))
+	}
+}
+
+// WithHTTPClient sets an http client
+func WithHTTPClient(c *http.Client) Option {
+	return func(args *Client) {
+		args.httpClient = c
+	}
+}
+
+// WithPerProviderTimeout sets the timeout when interacting with a BMC.
+// This timeout value is applied per provider.
+// When not defined and a context with a timeout is passed to a method, the default timeout
+// will be the context timeout duration divided by the number of providers in the registry,
+// meaning, the len(Client.Registry.Drivers).
+// If this per provider timeout is not defined and no context timeout is defined,
+// the defaultConnectTimeout is used.
+func WithPerProviderTimeout(timeout time.Duration) Option {
+	return func(args *Client) {
+		args.perProviderTimeout = func(context.Context) time.Duration { return timeout }
+	}
+}
+
+func WithIpmitoolCipherSuite(cipherSuite string) Option {
+	return func(args *Client) {
+		args.providerConfig.ipmitool.CipherSuite = cipherSuite
+	}
+}
+
+func WithIpmitoolPort(port string) Option {
+	return func(args *Client) {
+		args.providerConfig.ipmitool.Port = port
+	}
+}
+
+func WithIpmitoolPath(path string) Option {
+	return func(args *Client) {
+		args.providerConfig.ipmitool.IpmitoolPath = path
+	}
+}
+
+func WithAsrockrackHTTPClient(httpClient *http.Client) Option {
+	return func(args *Client) {
+		args.providerConfig.asrock.HttpClient = httpClient
+	}
+}
+
+func WithAsrockrackPort(port string) Option {
+	return func(args *Client) {
+		args.providerConfig.asrock.Port = port
+	}
+}
+
+func WithGofishHTTPClient(httpClient *http.Client) Option {
```

Comment:
As a library user with no clue what Gofish is, `Redfish` would be clearer in this case, since the user does not have to know what Gofish refers to. And while Gofish enables this library to connect to Redfish, I'm a bit conflicted with this change, but its not a blocker in anycase.

Reply:
this is a good point. i thought about using redfish but opted for gofish. I'm fine with using redfish. Will update.

Reply:
done